tests/tests_pytorch/loggers/test_mlflow.py

```diff
@@ -224,17 +224,28 @@ def test_mlflow_logger_with_unexpected_characters(client, _, __, tmpdir):
         logger.log_metrics(metrics)
 
 
+@mock.patch("pytorch_lightning.loggers.mlflow.Param")
 @mock.patch("pytorch_lightning.loggers.mlflow._MLFLOW_AVAILABLE", return_value=True)
 @mock.patch("pytorch_lightning.loggers.mlflow.MlflowClient")
-def test_mlflow_logger_with_long_param_value(client, _, tmpdir):
-    """Test that the logger raises warning with special characters not accepted by MLFlow."""
+def test_mlflow_logger_with_long_param_value(client, _, param, tmpdir):
+    """Test that the logger doesn't crash when logging a long parameter value."""
     logger = MLFlowLogger("test", save_dir=tmpdir)
-    value = "test" * 100
+    value = "test" * 200
     key = "test_param"
     params = {key: value}
 
-    with pytest.warns(RuntimeWarning, match=f"Discard {key}={value}"):
-        logger.log_hyperparams(params)
+    logger.log_hyperparams(params)
+
+
+@mock.patch("pytorch_lightning.loggers.mlflow.Param")
+@mock.patch("pytorch_lightning.loggers.mlflow._MLFLOW_AVAILABLE", return_value=True)
+@mock.patch("pytorch_lightning.loggers.mlflow.MlflowClient")
+def test_mlflow_logger_with_many_params(client, _, param, tmpdir):
+    """Test that the logger doesn't crash when logging more than 100 parameters."""
+    logger = MLFlowLogger("test", save_dir=tmpdir)
+    params = {f"test_param_{idx}": f"test_value_{idx}" for idx in range(200)}
+
+    logger.log_hyperparams(params)
```

Comment:
range(200) and logging in batch of 100 should result in two calls, right? Let's add an assertion like this:

assert client.log_batch.call_count == 2

Reply:
I think I've got a better idea of mock testing now. I added test for both mock calls.